builder/table_builder.go

```diff
@@ -24,7 +38,7 @@ func BuildTable(db *sql.DB) ([]string, []string) {
 		scanArgs[i] = &values[i]
 
 	}
-	var row = []string{}
+	var row = make([]string, 0)
```

Comment:
I want to refer to this recommendation https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices. I added `make` because the code wants to add elements to the slice; let me know what you think.

Reply:
I don't understand why you put make in front of my empty slice even after looking at the reference, could you explain it a bit better to me?

Reply:
I will try to explain my intention in a bit more detail.

Both `t := []string{}` and `t := make([]string, 0)` create an empty slice of type `[]string`. The difference is that `t := []string{}` creates an empty slice with no underlying array, whereas `t := make([]string, 0)` creates a slice with an underlying array of size 0.

The previous form, `t := []string{}`, creates a slice that points to a nil underlying array. This means that the slice has a length of 0 and a capacity of 0, and it does not have any memory allocated for storing elements. In other words, the slice is empty and does not have any capacity to store elements.

The changed form, `t := make([]string, 0)`, creates a slice with an underlying array of size 0. This means that the slice has a length of 0 but a capacity of at least 0, meaning it has some memory allocated for storing elements. 

In most cases, the difference between these two forms is unimportant and can be used interchangeably. However, you need to append elements to the slice later; in that case, you may prefer to use `t := make([]string, 0)` so that the underlying array is already allocated and doesn't need to be reallocated as you append elements.

Reply:
Thank you for elaborating, I understand it now.